### PR TITLE
Fix for #14

### PR DESCRIPTION
**Pull request by onetoomany**
changed dt_util.get_time_zone to dt_util.async_get_time_zone. Credit to @ma9mwah

### CODE REVIEW

**Review by BJReplay — COMMENTED** [review]
Have you tested this and confirm it works?

Sorry for the simple question, but I'm not a python programmer (but I'm trying to learn) nor do I know how to test changes like this (but I'm trying to learn).

Thanks for your PR and I hope to approve it shortly.

**Comment by fboundy** [comment]
I should be able to test this. Just need to get a few things set up.

File: custom_components/solcast_solar/__init__.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         entry.options[CONF_API_KEY],
         SOLCAST_URL,
         hass.config.path('solcast.json'),
-        dt_util.get_time_zone(hass.config.time_zone),
+        dt_util.async_get_time_zone(hass.config.time_zone),
```
Comment (luuuis) [inline custom_components/solcast_solar/__init__.py:94]: This should be:

```
        await dt_util.async_get_time_zone(hass.config.time_zone),
```

**Review by luuuis — CHANGES_REQUESTED** [review]

File: custom_components/solcast_solar/__init__.py (hunk 1 above)
Comment (fboundy) [reply custom_components/solcast_solar/__init__.py:94]: Agreed

**Review by fboundy — CHANGES_REQUESTED** [review]

**Comment by BJReplay** [comment]
Will this work on versions prior to 2024.6?

I get the impression from the blog article that the async method is new in .6

**Comment by BJReplay** [comment]
OK, so this doesn't work if you're not running 2024.6:

I'm running 2024.5.5 (docker).  As of yesterday, that was the latest version.

I simply edited __init__.py in my custom_components directory in place, and restarted HA.

```
homeassistant  | 2024-06-08 10:15:59.393 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry Solcast Solar for solcast_solar
homeassistant  | Traceback (most recent call last):
homeassistant  |   File "/usr/src/homeassistant/homeassistant/config_entries.py", line 584, in async_setup
homeassistant  |     result = await component.async_setup_entry(hass, self)
homeassistant  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
homeassistant  |   File "/config/custom_components/solcast_solar/__init__.py", line 94, in async_setup_entry
homeassistant  |     await dt_util.async_get_time_zone(hass.config.time_zone),
homeassistant  |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
homeassistant  | AttributeError: module 'homeassistant.util.dt' has no attribute 'async_get_time_zone'
```

I've had a look, and I can't see (but, as I've said, I'm brand new to HA and python) how to find out what version of HA an integration is running in).

So, unless / until someone can suggest how to either code this as a try to try both async or blocking, or depending on which version of HA we're running under call either async or blocking, I'd suggest leaving it as is (blocking), as this change will break the integration for anyone not running 2024.6.

**Comment by danielp370** [comment]
Hi @BJReplay thanks for taking over solcast-solar. Fearless leader.  
You can force the integration to require a particular HA version via hacs.json. See https://hacs.xyz/docs/publish/start/
So basically you'd need to update   "homeassistant": "2023.7" to  "homeassistant": "2024.6" if that is the path you want to go down.

**Comment by BJReplay** [comment]
> You can force the integration to require a particular HA version

@danielp370 thanks!

Ok, I will wait for others to comment, but I have now updated to 2024.6 and I think pretty much every integration threw that warning (some, multiple times).

My inclination is to reject the PR (or at least sit on it) as it is a breaking change for anyone unable to upgrade to 2024.6, and hold off for a few releases until there is much less likelihood that users will be on an old version.

I welcome advice from others who are far more knowledgeable than I am.

**Comment by changeofaspect** [comment]
The warning message reported by the original poster is exactly the same as what I am getting on my set up. 

```
Logger: homeassistant.util.loop
Source: util/loop.py:84
First occurred: 03:47:27 (6 occurrences)
Last logged: 11:14:50

Detected blocking call to open inside the event loop by custom integration 'solcast_solar' at custom_components/solcast_solar/solcastapi.py, line 97: with open(self._filename, "w") as f: (offender: /config/custom_components/solcast_solar/solcastapi.py, line 97: with open(self._filename, "w") as f:), please create a bug report at https://github.com/BJReplay/ha-solcast-solar/issues Traceback (most recent call last): File "<frozen runpy>", line 198, in _run_module_as_main File "<frozen runpy>", line 88, in _run_code File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module> sys.exit(main()) File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main exit_code = runner.run(runtime_conf) File "/usr/src/homeassistant/homeassistant/runner.py", line 190, in run return loop.run_until_complete(setup_and_run_hass(runtime_config)) File "/usr/local/lib/python3.12/asyncio/base_events.py", line 672, in run_until_complete self.run_forever() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 639, in run_forever self._run_once() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1988, in _run_once handle._run() File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run self._context.run(self._callback, *self._args) File "/usr/src/homeassistant/homeassistant/core.py", line 2741, in async_call response_data = await coro File "/usr/src/homeassistant/homeassistant/core.py", line 2784, in _execute_service return await target(service_call) File "/config/custom_components/solcast_solar/__init__.py", line 140, in handle_service_update_forecast await coordinator.service_event_update() File "/config/custom_components/solcast_solar/coordinator.py", line 67, in service_event_update await self.solcast.http_data(dopast=False) File "/config/custom_components/solcast_solar/solcastapi.py", line 543, in http_data await self.serialize_data() File "/config/custom_components/solcast_solar/solcastapi.py", line 97, in serialize_data with open(self._filename, "w") as f:
```

This occurs with __init__.py line 94  _dt_util.get_time_zone(hass.config.time_zone),_   but it also occurs with with
_await dt_util.async_get_time_zone(hass.config.time_zone),_

The warning message shows up every time that I poll Solcast server and I see it refers to the offender
```
(offender: /config/custom_components/solcast_solar/solcastapi.py, line 97: with open(self._filename, "w") as f:)
```

The snippet of code in the solastapi.py file looks like this:-
Solcastapi line 97


I am not a Python coder and don't really know what I am looking at or how to fix it but the warning message is clearly pointing to something wrong with this bit of code. Can someone with python skills take a look at this please as I am not sure the simple await dt_util.async_get_time_zone(hass.config.time_zone) change actually fixes the original reported problem.

**Comment by BJReplay** [comment]
OK,

There are several issues here:

Let's not fall into the trap of making one PR the placeholder for every issue that might sound the same - that's not how you do development or issue management or PRs.

Secondly, this is a warning, not an error.

Finally, until every user of this repo is on 2024.6, changing code to use async methods will break their use of this repo, so that's not an option.

I know we can make this repo depend on 2024.6, but that doesn't help anyone who can't upgrade for whatever reason.

I welcome suggestions on how we can know when is the right time to depend on 2024.6 so we can make changes to switch to async methods, but, for the moment, my view is that we simply ignore these warnings, and wait for others who are smarter python coders show us how they work around these warnings on much more widely used integrations, and then we can copy-pasta the changes, and use their example of how to make these breaking changes non-breaking for users.

**Comment by BJReplay** [comment]
Note - I'm leaving the conversation open, and if contributors have solutions to make this change non-breaking for users of earlier versions of HA, please amend the PR and re-open.

**Pull request closed without merging**